pandas/core/arrays/categorical.py

```diff
@@ -413,7 +413,10 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
             return array(self, dtype=dtype, copy=copy)
         if is_integer_dtype(dtype) and self.isna().any():
             raise ValueError("Cannot convert float NaN to integer")
-        return np.array(self, dtype=dtype, copy=copy)
+
+        new_categories = self.categories.astype(dtype)
```

Comment:
add a comment here on why we are doing this

Reply:
Done